frontend/src/Components/ShipsPlacement.js

```diff
@@ -0,0 +1,160 @@
+import { useNavigate } from 'react-router-dom';
+import styles from './Styles/ShipsPlacement.css';
+import Alert from '@mui/material/Alert';
+import Grid from '@mui/material/Grid';
+import CircularProgress from '@mui/material/CircularProgress';
+
+
+function ShipsPlacement(props) {
+    const navigate = useNavigate();
+    const [choosenShipInd, setChoosenShipInd] = useState('');
+    const [deleteShipFail, setDeleteShipFail] = useState(false);
+    const [placeShipFail, setPlaceShipFail] = useState('');
+    const [startGameFail, setStartGameFail] = useState(false);
+    const [shipRowCol, setShipRowCol] = useState({start: [], end: []});
+    const [isLoading, setIsLoading] = useState(false);
+    const alphabet = ["A","B","C","D","E","F","G","H","I","J","K","L","M","N","O","P","Q","R","S","T","U","V","W","X","Y","Z"];
+
+
+    function getIndexesData(rowColData) {
+        if (shipRowCol.start.length === 0) {
+            setShipRowCol(prev => ({
+                ...prev,
+                start: [(rowColData[0]), (rowColData[1])]
+            }));
+        } else {
+            setShipRowCol(prev => ({
+                ...prev,
+                end: [(rowColData[0]), (rowColData[1])]
+            }));
+        }
+    };
+
+
+    async function placeShip() {
+        if (shipRowCol.end.length !== 0) {
+            setIsLoading(true);
+            try {
+                const newShipData = {
+                    ship_size: Number(props.remainingShips[choosenShipInd]),
+                    start_row: shipRowCol.start[0],
+                    start_col: shipRowCol.start[1],
+                    end_row: shipRowCol.end[0],
+                    end_col: shipRowCol.end[1]
+                }
+                const result = await placeAShip(props.game_id, newShipData);
+                if (result.status === 200) {
+                    props.getTheGameInfo();
+                    setIsLoading(false);
+                } else {
+                    const jsonData = await result.json();
+                    setPlaceShipFail(jsonData.msg);
+                    setShipRowCol({start: [], end: []});
+                    setChoosenShipInd('');
+                    setIsLoading(false);
+                }
+            } catch (e) {
+                navigate('/error');
+            }
+        }
+    };
+
+    function handleChoosenShipChange(e, i) {
+        setChoosenShipInd(i);
+    };
+
+    async function deleteShip(e) {
+        e.preventDefault();
+        setIsLoading(true);
+        try {
+            const result = await deleteAShip(props.game_id, props.placedShips[e.target.value]);
+            if (result === true) {
+                props.getTheGameInfo();
+                setIsLoading(false);
+            } else {
+                setDeleteShipFail(true);
+                setIsLoading(false);
+            }
+        } catch(e) {
+            navigate('/error');
+        }
+    };
+    
+
+    useEffect(() => {
+        placeShip();
```

Comment:
it is better to move the condition 
```
if (shipRowCol.end.length !== 0) {
```
from within the function `placeShip` to the body of the `useEffect`, in order to avoid a call to an `async` function if not necessary.